### PR TITLE
refactor: apply DisallowedEmptyRuleFixerRector

### PR DESCRIPTION
**Pull request by kenjis**
~~Needs #8341~~

**Description**
- replace `empty()`
- add and apply
   - TernaryEmptyArrayArrayDimFetchToCoalesceRector
   - EmptyOnNullableObjectToInstanceOfRector
   - DisallowedEmptyRuleFixerRector

**Checklist:**
- [x] Securely signed commits
- [] Component(s) with PHPDoc blocks, only if necessary or adds value
- [] Unit testing, with >80% coverage
- [] User guide updated
- [x] Conforms to style guide

### CODE REVIEW

File: system/Common.php [hunk 1]
```diff
@@ -1121,15 +1125,16 @@ function stringify_attributes($attributes, bool $js = false): string
      * returns its return value if any.
      * Otherwise will start or stop the timer intelligently.
      *
+     * @param non-empty-string|null    $name
      * @param (callable(): mixed)|null $callable
      *
-     * @return Timer
+     * @return mixed|Timer
```
Comment (paulbalandan) [inline system/Common.php:1131]: We can use conditional return here.

File: system/Test/FeatureTestCase.php [hunk 2]
```diff
@@ -371,16 +371,19 @@ protected function setRequestBody($request, ?array $params = null)
...
             $formatMime = '';
+
             if ($this->bodyFormat === 'json') {
                 $formatMime = 'application/json';
             } elseif ($this->bodyFormat === 'xml') {
                 $formatMime = 'application/xml';
             }
-            if (! empty($formatMime) && ! empty($params)) {
+
+            if ($formatMime !== '' && ! empty($params)) {
```
Comment (paulbalandan) [inline system/Test/FeatureTestCase.php:386]: missed empty

File: system/Common.php [hunk 3]
```diff
@@ -1121,15 +1125,17 @@ function stringify_attributes($attributes, bool $js = false): string
...
+     * @phpstan-return ($callable is (callable(): mixed) ? mixed : Timer)
```
Comment (paulbalandan) [inline system/Common.php:1132]: I think this should be based on `$name`?

Comment (kenjis) [reply system/Common.php:1132]: See the code. `stop()` and `start()` return `$this`.
https://github.com/codeigniter4/CodeIgniter4/blob/64866aa3c46a4e9bfc484e9da850b14329865e7f/system/Common.php#L1134

Comment (paulbalandan) [reply system/Common.php:1132]: Okay. Then based on `$name` and `$callable`.

Comment (kenjis) [reply system/Common.php]: Oh, you are correct. Fixed.

**Review by paulbalandan — APPROVED** [review]

**Review by michalsn — APPROVED** [review]

**Pull request merged**